index.js

```diff
@@ -112,48 +112,29 @@ function EncryptionStream(options) {
     return new EncryptionStream(options);
   }
 
-  // It stinks the we have to store all the
-  // encrypted chunks in memory but such is life
-  this._cipherTextChunks = [];
-
-  this._started = false;
-  this._nonce = options.nonce || exports.createSalt(12);
+  var nonce = options.nonce || exports.createSalt(12);
 
   if (validateKey(options.key)) {
     this._key = options.key;
-    this._cipher = crypto.createCipheriv('aes-256-gcm', this._key, this._nonce);
+    this._cipher = crypto.createCipheriv('aes-256-gcm', this._key, nonce);
   }
 
   Transform.call(this, options);
+  this.push(nonce);
 }
 util.inherits(EncryptionStream, Transform);
 
 EncryptionStream.prototype._transform = function(chunk, enc, cb) {
-  if (!this._started) {
-    this._started = true;
-    this.push(this._nonce);
-  }
-
-  this._cipherTextChunks.push(this._cipher.update(chunk));
-
-  if (typeof(cb) === 'function') {
-    cb();
-  }
+  this.push(this._cipher.update(chunk));
+  cb();
 };
 
 EncryptionStream.prototype._flush = function(cb) {
   // final must be called on the cipher before generating a MAC
-  this._cipherTextChunks.push(this._cipher.final());
-  var mac = this._cipher.getAuthTag();
-  this.push(mac); // 16 bytes
+  this._cipher.final(); // this will never output data
```

Comment:
While I agree in practice it never outputs data the docs make it seem like it could:

```
Returns any remaining enciphered contents.
```

No harm to also push it into the stream